src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        boolean isHeader = true; // Skip the header line
+        for (String line : data) {
+            if (isHeader) {
+                isHeader = false;
+                continue;
+            }
+            String[] parts = line.split(",");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid data format");
```

Comment:
The error message 'Invalid data format' is not very informative. It would be helpful to include more context, such as the line that caused the error, so the user understands what exactly is wrong with the data format.